packages/react/src/components/Comments/ThreadsSidebar.tsx

```diff
@@ -0,0 +1,188 @@
+import { useBlockNoteEditor } from "../../hooks/useBlockNoteEditor.js";
+import { useUIPluginState } from "../../hooks/useUIPluginState.js";
+import { Thread } from "./Thread.js";
+import { useThreads } from "./useThreads.js";
+import { ThreadData } from "@blocknote/core/comments";
+import { BlockNoteEditor } from "@blocknote/core";
+import { useEffect, useMemo } from "react";
+import { TextSelection } from "@tiptap/pm/state";
+
+function sortThreads(
+  editor: BlockNoteEditor<any, any, any>,
+  threads: ThreadData[],
+  sort?: "position" | "newest" | "oldest"
+) {
+  if (sort === "position") {
+    return threads.sort((a, b) => {
+      const threadA =
+        editor.comments?.plugin
+          .getState(editor.prosemirrorState)
+          .threadPositions.get(a.id)?.from || Number.MAX_VALUE;
+      const threadB =
+        editor.comments?.plugin
+          .getState(editor.prosemirrorState)
+          .threadPositions.get(b.id)?.from || Number.MAX_VALUE;
+
+      return threadA - threadB;
+    });
+  }
+
+  if (sort === "newest") {
+    return threads.sort((a, b) => {
+      return b.createdAt.getTime() - a.createdAt.getTime();
```

Comment:
https://github.com/TypeCellOS/BlockNote/pull/1454#issuecomment-2721015972
```suggestion
      const aLastComment = a.comments.reduce((acc, comment) => {
        return comment.createdAt > acc.createdAt ? comment : acc;
      }, a.comments[0]);
      const bLastComment = b.comments.reduce((acc, comment) => {
        return comment.createdAt > acc.createdAt ? comment : acc;
      }, b.comments[0]);
      return bLastComment.createdAt.getTime() - aLastComment.createdAt.getTime();
```

Reply:
very curious why you prefer that above sort?

Reply:
Let's say that I got a notification that a review was done on my document that already had a bunch of existing comments on them. If I had sorted the list by recency, I would expect to just review those items one by one, until I hit the item that was no longer apart of this review anymore.

Whereas, to see that a new comment happened in this sort, I'd have to dig through each thread to see which comment happened and whether I need to action on it or not.

Like, this is sorted by "newest", but especially since we only show to the granularity of a day, it actually feels like a random sort:
<img width="371" alt="image" src="https://github.com/user-attachments/assets/a08cd526-d194-42a3-ad85-577d9fd7d090" />

Reply:
Can't we just do this instead?
```ts
threads.sort(
  (a, b) =>
    b.comments[b.comments.length - 1].createdAt.getTime() -
    a.comments[a.comments.length - 1].createdAt.getTime()
);
```
Since new comments are pushed to the end of the array, do we need to check all of them or just the last one in the array?

Reply:
Yea, probably fine. I did it against all of them to capture edits, but that is likely good enough

Reply:
fyi, I think gdocs simply sorts by thread creation time. In this case I think we could just keep it simple next time; probably both solutions are somewhat confusing to the end user. Best way to address nicks concerns is to implement notifications down the line ;)

for now let's keep what we have in the codebase